### PR TITLE
ClusterControl.joined should cause joining for simpler usage

### PR DESCRIPTION
**Pull request by ktoso**
This hardens singleton handover, sadly it also hits a bug in IRGen which made the impl uglier than I would have wanted to.

This is not to be reviewed yet, I want to give it some runs on CI

### CODE REVIEW

File: Sources/DistributedCluster/Backoff.swift [hunk 1]
```diff
@@ -208,6 +206,39 @@ public struct ExponentialBackoffStrategy: BackoffStrategy {
...
+    /// Attempt to execute the passed `operation` at most `maxAttempts` times while applying the expected backoff strategy.
+    ///
+    /// - Parameters:
+    ///   - operation: The operation to run, potentially multiple times until successful or maxAttempts were made
+    /// - Throws: the last error thrown by `operation`
+    public func attempt<Value>(_ operation: () async throws -> Value) async throws -> Value {
+        var backoff = self
+        var lastError: Error? = nil
+        defer {
+            pprint("RETURNING NOW singleton")
```
Comment (yim-lee) [inline Sources/DistributedCluster/Backoff.swift:218]: This and L224 probably not intended to keep

File: Sources/DistributedCluster/DeadLetters.swift [hunk 2]
```diff
@@ -220,6 +220,10 @@ public final class DeadLetterOffice {
                 return metadata
             }()
         )
+
+        pinfo("ALL ACTORS: \(self.system?.cluster.node)")
+        self.system?._printTree()
+        print("\n\n======")
```
Comment (yim-lee) [inline Sources/DistributedCluster/DeadLetters.swift:226]: intended to keep?

**Review by yim-lee — COMMENTED** [review]
I _think_ changes in this PR are already covered by https://github.com/apple/swift-distributed-actors/pull/1127 and https://github.com/apple/swift-distributed-actors/pull/1122?

**Comment by ktoso** [comment]
Oh whoops, I did something wrong here with PRs. Let me check if there is anything missing from main still

**Comment by ktoso** [comment]
You're right my bad, this was subsumed already.

**Pull request closed without merging**